astra/src/main/java/com/slack/astra/server/Astra.java

```diff
@@ -223,7 +223,10 @@ private static Set<Service> getServices(
               .withRequestTimeout(requestTimeout)
               .withTracing(astraConfig.getTracingConfig())
               .withAnnotatedService(new ElasticsearchApiService(astraDistributedQueryService))
-              .withAnnotatedService(new ZipkinService(astraDistributedQueryService))
+              .withAnnotatedService(
+                  new ZipkinService(
+                      astraDistributedQueryService,
+                      astraConfig.getQueryConfig().getZipkinDefaultMaxSpans()))
```

Comment:
I think it'd be good to add a precondition for the zipkinDefaultMaxSpans value. Maybe add another checkArgument call for it in `ValidateAstraConfig.validateQueryConfig`?